sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodCheck.java

```diff
@@ -0,0 +1,262 @@
+        this.methodName = CommonUtils.createPattern(methodName);
+    }
+
+    /**
+     * Set number or range to match number of arguments of the forbidden method.
+     * Multiple values must be comma separated.
+     * @param argumentCount range for matching number of arguments
+     * @throws CheckstyleException when argumentCount is not a valid range
+     */
+    public void setArgumentCount(String argumentCount) throws CheckstyleException {
+        this.argumentCount = argumentCount;
+        if (argumentCount.trim().length() > 0) {
+            final String[] rangeTokens = COMMA_REGEX.split(argumentCount);
+            for (String oneToken : rangeTokens) {
+                this.argumentCountRanges.add(new IntRange(oneToken));
+            }
+        }
+        else {
+            throw new CheckstyleException(
+                "argumentCount must be non-empty, found: " + argumentCount);
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL:
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                final String methodNameInCode;
+                if (dot.getType() == TokenTypes.IDENT) {
+                    methodNameInCode = dot.getText();
+                }
+                // method that looks like: obj.method()
+                else {
+                    methodNameInCode = dot.getLastChild().getText();
+                }
+                final int numArgsInCode = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, methodNameInCode, numArgsInCode);
+                break;
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Check if the method/constructor call against defined rules.
+     * @param ast ast for the call
+     * @param name ruleName of the the method
+     * @param argCount number of arguments of the method
+     */
+    private void checkForbiddenMethod(DetailAST ast, String name, int argCount) {
+        if (methodName.matcher(name).matches()) {
+            if (argumentCountRanges.isEmpty()) {
+                log(ast.getLineNo(), ast.getColumnNo(), MSG_KEY_WITHOUT_ARG, name, methodName);
+            }
+            else {
+                for (IntRange intRange : argumentCountRanges) {
+                    if (intRange.contains(argCount)) {
+                        log(ast.getLineNo(), ast.getColumnNo(), MSG_KEY_WITH_ARG, name, methodName,
```

Comment:
if range of arguments always exists , there is not need  to make separate "log" messages.